python/mxnet/gluon/contrib/estimator/estimator.py

```diff
@@ -88,12 +88,17 @@ def _check_metrics(self, metrics):
         return metrics
 
     def _check_context(self, context):
-        # handle context
-        if isinstance(context, Context):
-            context = [context]
-        elif isinstance(context, list) and all([isinstance(c, Context) for c in context]):
-            context = context
-        elif not context:
+        if context:
+            # check context values, only accept Context or a list of Context
+            if isinstance(context, Context):
+                context = [context]
+            elif isinstance(context, list) and all([isinstance(c, Context) for c in context]):
+                context = context
```

Comment:
should we check for the GPU device index too? also, try querying `num_gpus()` only once.

Reply:
@szha thanks! I m now asserting context must be in `available_context` which is `[cpu(), gpu(0), ...., gpu(num_gpus-1)]`.

Reply:
cpu(65536) is actually valid in mxnet regardless of the number of physical CPUs

Reply:
How about:
```
available_context = [mx.gpu(i) for i in range(num_gpus)]
assert ctx  in available_context or str(ctx).startswith('cpu')
```